include/cantera/thermo/MixtureFugacityTP.h

```diff
@@ -533,11 +517,22 @@ class MixtureFugacityTP : public ThermoPhase
 
     virtual void updateMixingExpressions();
 
+
+    //! Solve the cubic equation of state
+    /*!
+     * 
+     * Returns the number of solutions found. If it only finds the liquid
+     * branch solution, it will return a -1 or a -2 instead of 1 or 2.  If it
+     * returns 0, then there is an error.
+     * The cubic equation is solved using Nickall's method (Ref: The Mathematical Gazette(1993), 77(November), 354�359, https://www.jstor.org/stable/3619777)
+     */
+    int NicholsSolve(double T, double pres, double a, double b, 
+                                        double aAlpha, double Vroot[3], double an,
+                                        double bn, double cn, double dn, double tc) const;
```

Comment:
I've often wondered why it was named Nichols.  Would be in favor of renaming it, so long as there is no fear we would ever want to maintain multiple cubic solvers in this class.  Hard to envision a use case, but one never knows...

Reply:
@decaluwe If we ever want to add multiple solvers, we can do that pretty easily if we have a generic method called `solveCubic()`, which could be changed to just call the appropriate specific solver at that time, with no other changes needed (at least, ideally).

Reply:
@bryan, yes that was my thinking… but then the `solverCubic()` would be deployed at the child class level, and this would be a pass-through to the specific solver, no? I.e. the methods in `Peng-Robinson` would call `solverCubic()`, as needed, and this would setup and call the specific solver (in this case, still called something like `NichalsCubic()`).

Reply:
I think if you have a `solveCubic()` in `MixtureFugacityMFTP`, then individual subclasses can override that as necessary, but not every subclass needs to override it. Even if the method is defined in `MixtureFugacityMFTP`, subclasses can call it with no problem (you might need to make it `virtual`, we're not at the limit of my C++ knowledge. @speth can help with what specifically needs to be done). The point is that I think we can rename this method without causing any problems in the future.

Reply:
Okay, this is fixed by https://github.com/Cantera/cantera/commit/095ee5a912eb19770cd9c802202885f065b3fd87. I converted the `NicholsCall()` function to `cubicSolve()`, which sets up the necessary variables and then calls `MixtureFugacity::cubicSolve()`.